hammers/query.py

```diff
@@ -232,6 +228,44 @@ def owned_compute_ip_single(db, project_id):
     '''.format(projcol=project_col(db.version))
     return db.query(sql, args=[project_id], limit=None)
 
+@query
+def idle_floating_ips(db, threshold_days):
+    '''
+    Return all idle floating ips who satisfy the following conditions
```

Comment:
You can add the reservation check as a criteria here. Blazar will add a tag to any floating ips it controls. I think adding a join would work, something like

```mysql
LEFT JOIN neutron.tags AS t
ON f.standard_attr_id = t.standard_attr_id
WHERE (t.tag IS NULL OR (t.tag != 'blazar' AND t.tag NOT LIKE 'reservation:%'))
```

Reply:
Thanks, @diurnalist! Good to know :) I fixed it and tested. Will merge.